docs/docs/07-examples/10-using-javascript.md

```diff
@@ -4,7 +4,78 @@ id: using-javascript
 keywords: [example, javascript, extern, typescript, js, ts]
 ---
 
-Calling a Javascript function from Wing requires two steps.
+## Creating inflight function from JavaScript/TypeScript file
+
+Inflight closures are an extremely important Wing functionality. Consider the following simple wing program:
+
+```wing
+// main.w
+bring cloud;
+let bucket = new cloud.Bucket();
+
+bucket.onCreate(inflight (file) => {
+  log(file);
+});
+```
+
+Being able to write inflight wing alongside the preflight code is beautiful, but you may want to write the inflight function in a separate file and *language*. The `@inflight` intrinsic function can be used to create an inflight closure from a JavaScript/TypeScript:
+
+```wing
+// main.w
+bring cloud;
+let bucket = new cloud.Bucket();
+
+bucket.onCreate(@inflight("./bucket_create.ts"));
+//              ^ onCreate expects an `inflight (str): void` function, so the file must export a function with a typescript signature that matches
+//                        ^ Relative (to current file) path to javascript or typescript file
+//                          Note: This must be a static string
+```
+
+`wing compile` will generate `bucket_create.inflight.ts` while will which will contain all of the information needed for TypeScript type checking and IDE support.
+With that, you can create the `bucket_create.ts` file:
+
+```ts
+// bucket_create.ts
+import inflight from "./bucket_create.inflight";
```

Comment:
@MarkMcCulloh , there is something counter intuitive for me in this code:

```ts
// bucket_create.ts
import inflight from "./bucket_create.inflight";

export default inflight(async ({ myQueue }, file) => {
```


Can you explain why do we need an inflight to wrap the actual function?

Reply:
The primary reason is a DX improvement, there is currently no *functional* purpose for this since the function just returns the argument.

The dilemma here is that typescript (and javascript) don't provide an easy mechanism for "Given type X, make sure this file exports it".

I think the current extern mechanism suffers from this, so you have to write code like this:

```ts
import extern from "./blah.extern";

export const func: extern["func"] = (name) => {
  return `Hello, ${name}!`;
};
```

I think the `inflight()` function helps guide user behavior, although the function type is also exported so user's can choose to use the explicit typing approach as well.